packages/website/docs/contributing/guides/1_adding-a-component.mdx

```diff
@@ -0,0 +1,457 @@
+
+```mermaid
+flowchart TB
+    q{{What is your component's role in a data visualization?}}
+    q --> a["The main component. It renders the data
+    directly and could stand alone"] --> Core([Core])
+    q --> b[It is meant to accompany other components] --> Auxiliary([Auxiliary])
+    Core --> data{Data type}
+    data -- Series --> Component([Core Component])
+    data -- Graph --> Component
+    data -- MapGraph --> Component
+    data -- Other --> Component
+    Component --> SVG{SVG Supported?}
+    SVG -- yes --> SVGComponent(["ComponentCore
+    type: SVG"])
+    SVG -- no --> HTMLComponent(["ComponentCore
+    type: HTML"])
+    data -- XYData --> XYComponent([XY Component])
+    Auxiliary --> auxXY{XY data?}
+    auxXY -- yes --> XYComponent
+    auxXY -- no --> StandAlone[Stand-alone]
+```
+
+## Component code
+### File Structure
+Your component will exist in the `@unovis/ts` library. Once you decide on a name for the component,
+create a new directory inside `src/components` called _your-component-name_. Note that we use kebab
+case naming convention for all files in this package.
+
+The component directories follow this structure:
+```yaml
+component-name
+├── modules
+│   └── # any extraneous logic/helper functions belong here in separate files
+├── config.ts # ┓
+├── index.ts  # ┠ core files
+├── style.ts  # ┛
+├── types.ts # any component-specific custom types go here
+```
+
+### Configuration
+The logic for component configuration is contained in `config.ts`. This file defines the configuration
+properties and sets default values for those properties.
+The general structure looks like:
+
+```ts title=config.ts
+export interface ConfigInterface {
+  /** Property JSDoc. Default value: `` */
+}
+export class Config implements ConfigInterface {
+  // Default values
+}
+```
+
+In most cases (i.e. for any non stand-alone component), _Config_ and _ConfigInterface_ should also extend
+their corresponding super classes. (`ComponentConfig` and `ComponentConfigInterface` for Core components,
+`XYComponentConfig` and `XYComponentConfigInterface` for XY components).
+
+### Selectors and Styling
+The `style.ts` file declares selectors and sets default styles. Think of selectors as the
+structural building blocks that make up your component. (For example, _Scatter_ contains a selector
+labeled "point", each of which are  rendered as a SVG `circle` elements and have various SVG properties).
+Any CSS variables will also be initialized in this file.
+
+For style composition use the following methods from [@emotion](https://emotion.sh/docs/@emotion/css):
+_injectGlobal_ for CSS variables and _css_ to initialize unique class name for each selector.
+
+The general template for a `style.ts` file looks like:
+
+```ts title=style.ts
+import { css, injectGlobal } from '@emotion/css'
+
+export const root = css`
+ label: component-name;
+`
+export const globalStyles = injectGlobal`
```

Comment:
I think it's work providing examples according to the new way of defining variables (see `LeafletMap`)